From 878be921d1a6ceb6758bc3c1efbd041897566c42 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Olivier=20De=20Canni=C3=A8re?= <olivier.decanniere@qt.io>
Date: Thu, 16 Oct 2025 09:21:05 +0200
Subject: [PATCH 180/239] Compiler: Actually read the value of a key passed to
 qmltypes

We always set isJavaScriptBuiltin to true if the key was present instead
of actually reading the value. This seems to have worked because the key
was only added when it was true and the default value is false.

Amends 578946437fb31edd6aa78612e9db8fed9a3eb210

Pick-to: 6.8
Change-Id: I91f533129d233b251e221bf05abdc919eb8957c6
Reviewed-by: Sami Shalayel <sami.shalayel@qt.io>
Reviewed-by: Fabian Kosmale <fabian.kosmale@qt.io>
(cherry picked from commit 72e12635d81cd86362a3eaf2951160550c3a3ab1)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/qmlcompiler/qqmljstypedescriptionreader.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/qmlcompiler/qqmljstypedescriptionreader.cpp b/src/qmlcompiler/qqmljstypedescriptionreader.cpp
index f973630e1b..53679977de 100644
--- a/src/qmlcompiler/qqmljstypedescriptionreader.cpp
+++ b/src/qmlcompiler/qqmljstypedescriptionreader.cpp
@@ -252,7 +252,7 @@ void QQmlJSTypeDescriptionReader::readComponent(UiObjectDefinition *ast)
             } else if (name == QLatin1String("immediateNames")) {
                 readImmediateNames(script, scope);
             } else if (name == QLatin1String("isJavaScriptBuiltin")) {
-                scope->setIsJavaScriptBuiltin(true);
+                scope->setIsJavaScriptBuiltin(readBoolBinding(script));
             } else {
                 addWarning(script->firstSourceLocation(),
                            tr("Expected only lineNumber, name, prototype, defaultProperty, "
-- 
2.51.2

